### PR TITLE
Show EPG image on the PVR screens

### PR DESCRIPTION
**Pull request by lightglitch**
## Description
Show EPG image on the PVR Home channels widget, Dialog Seek Bar during video playback, Dialog PVRChannels screen and MyPVRChannels screen

## How Has This Been Tested?
Tested on the lastest nightly

## Screenshots (if appropriate):
PVR Home Widget:
image

MyPVRChannels:
image

Dialog PVR Channels:
image
Dialog Seek Bar:
image

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by ksooo** [comment]
Not sure I like (all of) this changes. Give me some time to think about it.

**Comment by ksooo** [comment]
My feeling is that we should only display the epg event icon if the window/dialog displays "deep" details of the event (like the plot).

For instance I don't want the image of the playing epg event in the home screen widget, because the channel logo is the eye catcher here to find a channel. Without the logo, it is much more difficult to find a certain channel of interest. And the purpose of that widget is to list channels, not epg events. 

Generally, if the epg event is the spot of displayed information, then epg event icon is fine, if focus is on the channel, channel logo is to prefer.

Makes sense?

**Comment by ksooo** [comment]
So, +1 for
* MyPVRChannels
* Dialog Seek Bar

-1 for: 
* PVR Home Widget
* Dialog PVR Channels

**Comment by ksooo** [comment]
On second thought, -1 for Dialog Seek Bar as well, sorry.

Looking at it from a users perspective: My kids always use the channel logo to identify the channel they are currently watching or if in OSD cahnnel preview the channel they want to switch to.

**Comment by ksooo** [comment]
Hate to say this, but I'm only fine with the proposed change for MyPVRChannels.

**Comment by ksooo** [comment]
One last thought, maybe we find a way to display both channel logo and event image in the seek bar dialog - channel logo on the left, event image on the right? Just an idea.

**Comment by lightglitch** [comment]
I understand your points, I just don't agree with your position on the Dialog Seek Bar screen.
How easy would be to add some skin settings to control this options?

**Comment by ksooo** [comment]
No skin settings for this, please. Those kind of bells and whistles are for other skins, like Estuary mods. Estuary itself is ment to be simple and lightweight.

**Comment by ksooo** [comment]
>  maybe we find a way to display both channel logo and event image in the seek bar dialog - channel logo on the left, event image on the right?

 What's your opinion on this?

**Comment by lightglitch** [comment]
I think could be a good option, I going to add it and see how it looks.

**Comment by lightglitch** [comment]
image
Approved?

**Comment by ksooo** [comment]
I like it, but I'm not a designer. Let's ask somebody more experienced with UI design: @da-anda, what do you think about the "logo on the left, thumb on the right" approach shown in the screenshot above?

**Comment by da-anda** [comment]
phew, tough one. Will think about it and play around in a mockup to see if I can find a better spot

**Comment by da-anda** [comment]
oh, before I forget. I'm with ksooo on which sceens should show the EPG thumbnail. However, for PVR recordings on the home screen widget I'd prefer the EPG icon (if still available) or recording thumb if available. So for the currently recorded show in the recordings widget we could probably show the EGP icon. Just a thought.

**Comment by lightglitch** [comment]
l can put just for the recordings but I'm unable to test it because don't have recordings active in my system.

**Comment by ksooo** [comment]
@da-anda are you fine with the epg image on the right side and the channel logo on the left side of the OSD, like shown in the screenshot above? Or was this the written you wanted to think about a bit more?

**Comment by da-anda** [comment]
that's what I wanted to think about and I already played around with it in Photoshop, but haven't found a better spot. If you put it to the left side it's IMO too much, and we don't have many other options with current OSD design. So fine with me as it is (given that the alignments match - so top of image should be top of plot)

**Comment by lightglitch** [comment]
Ok, so change the alignment from center to top? Because now is center and because is keeping the aspect ratio we can't guarantee that they will be align.

**Comment by ksooo** [comment]
> Ok, so change the alignment from center to top? Because now is center and because is keeping the aspect ratio we can't guarantee that they will be align.

 No, this will definitely look ugly, especially if logo and thumb have not the same aspect ratio. Keep it as it is - center.

**Comment by lightglitch** [comment]
Update pull request

**Comment by sualfred** [comment]
> @lightglitch : l can put just for the recordings but I'm unable to test it because don't have recordings active in my system.

> @ksooo : Please let recordings out of this PR. We can discuss this separately. Not mixing in too many stuff into one PR helps me to stay focused.

FYI -> ListItem.EpgEventIcon is not available for recordings.

**Comment by ksooo** [comment]
Problem with recordings is that epg event images are not stored along with the recordings data. Thus, they will only be available as long as the data is in memory/epg database - which usually is a short timeframe only for epg events in the past.

If we want to provide those images for a recording's lifetime, we would need to extend PVR add-on API and the add-ons to support this.

This, I currently see not much sense in extending listitem.epgeventicon to support recordings.

**Comment by sualfred** [comment]
Yep, and another problem is that EPG thumbs are usually external web stored images. Noone knows how long the EPG provider stores this image on their servers.

**Comment by lightglitch** [comment]
@ksooo the pull request is updated only for the screens you mentioned before so it's ready to be approved if everything is ok.

File: addons/skin.estuary/xml/DialogSeekBar.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogSeekBar.xml]: Shouldn't we add a `fallback="DefaultTVShows.png"` to the texture tag? Otherwise we will have an ugly empty space at the right if no epg image is avalaible.  Then, the visible condition can be reduced to `<visible>VideoPlayer.HasEpg</visible>`

Comment (lightglitch) [reply addons/skin.estuary/xml/DialogSeekBar.xml]: I think in this case the empty space is better than a TV icon that is not related with anything in the bar. I would prefer an empty space.

Comment (lightglitch) [reply addons/skin.estuary/xml/DialogSeekBar.xml]: @ksooo Here are the screens.
Empty space:
image
TV Icon:
image
Empty space all the way for me.

Comment (ksooo) [reply addons/skin.estuary/xml/DialogSeekBar.xml]: You need a longer, multi-line plot to see what I mean. Then it makes a difference,

Comment (ksooo) [reply addons/skin.estuary/xml/DialogSeekBar.xml]: IFAIK, in Kodi we always use fallback images if no thumbnail is avail.

Comment (ksooo) [reply addons/skin.estuary/xml/DialogSeekBar.xml]: Best case imo is to make the plot text field wider if no epg event icon is avail. That would be my number 1 candidate.

Comment (lightglitch) [reply addons/skin.estuary/xml/DialogSeekBar.xml]: I and tested it and saw what you mean by empty square. Made the changes.
I tried to do something like:
```
<variable name="VideoPlayerPlotRight">
	<value condition="VideoPlayer.HasEpg">260</value>
	<value>30</value>
</variable>
```
```
<right>$VAR[VideoPlayerPlotRight]</right>
```
But didn't work, still don't understand when and where can we use $VAR or not.

**Comment by lightglitch** [comment]
@ksooo How about this solution?

**Comment by ksooo** [comment]
The textbox labels have no width. Doesn't the epg image overlap very long plots? Despite from this, its looking good to me.

**Comment by ksooo** [comment]
Wait, they have 'right' tags. All fine, then.

**Comment by ksooo** [comment]
For better readability, I would swap the definitions of the second textbox and the epg image. If you think "left to right" the image comes after the textbox. But you only need to change is, if you like.

**Comment by lightglitch** [comment]
@ksooo done.

File: addons/skin.estuary/xml/DialogSeekBar.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogSeekBar.xml]: you can omit all three hasepg checks you have introduced here. They are not needed. The length  of the epgeventicon string is sufficient.

**Comment by lightglitch** [comment]
@ksooo updated.

**Comment by ksooo** [comment]
@ronie for final checkout?

**Review by ksooo — APPROVED** [review]
Thanks.

File: addons/skin.estuary/xml/Includes_PVR.xml
Comment (ronie) [inline addons/skin.estuary/xml/Includes_PVR.xml]: i thought (perhaps i'm wrong) that ListItem.Icon will already display the EPG Event Icon if available?

in case i'm wrong and there is a usecase for this change, please use a $VAR[] instead of duplicating the image control.

**Review by ronie — CHANGES_REQUESTED** [review]

File: addons/skin.estuary/xml/Includes_PVR.xml
Comment (ksooo) [reply addons/skin.estuary/xml/Includes_PVR.xml]: @ronie you are wrong. If it's a listitem for a channel (like in this case) there is no fallback to or preference of the event thumbnail. Only if it is a listitem for an epg event, there is a fallback to the channel logo if no epg thumb is available.

Comment (ksooo) [reply addons/skin.estuary/xml/Includes_PVR.xml]: (however invented this logic....)

**Review by ronie — APPROVED** [review]

**Pull request merged**
